Build failing on Windows

### What type of bug is this?

Unexpected error

### What subsystems and features are affected?

Build system

### What happened?

When I try to build the project on Windows following the indications of https://docs.timescale.com/self-hosted/latest/install/installation-source/  unexpected errors appear.



### TimescaleDB version affected

2.16.1

### PostgreSQL version used

16.4

### What operating system did you use?

Windows 10 Pro

### What installation method did you use?

Source

### What platform did you run on?

On prem/Self-hosted

### Relevant log output and stack trace

```bash
D:\GIT\TIMESCALE\timescaledb>cmake --build ./build --config Release
Versión de MSBuild 17.11.2+c078802d4 para .NET Framework

  1>Checking Build System
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.17.0-dev.sql
  Building Custom Rule D:/GIT/TIMESCALE/timescaledb/sql/CMakeLists.txt
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/updates/post-update.sql.processed
  Building Custom Rule D:/GIT/TIMESCALE/timescaledb/sql/CMakeLists.txt
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.10.0--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.10.1--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.10.2--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.10.3--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.11.0--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.11.1--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.11.2--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.12.0--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.12.1--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.12.2--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.13.0--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.13.1--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.14.0--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.14.1--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.14.2--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.15.0--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.15.1--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.15.2--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.15.3--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.16.0--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.16.1--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.5.0--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.5.1--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.5.2--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.6.0--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.6.1--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.7.0--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.7.1--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.7.2--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.8.0--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.8.1--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.9.0--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.9.1--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.9.2--2.17.0-dev.sql
  Generating D:/GIT/TIMESCALE/timescaledb/build/sql/timescaledb--2.9.3--2.17.0-dev.sql
  Building Custom Rule D:/GIT/TIMESCALE/timescaledb/src/CMakeLists.txt
  Building Custom Rule D:/GIT/TIMESCALE/timescaledb/src/CMakeLists.txt
  uuid.c
  func_cache.c
  cache.c
  cache_invalidate.c
  chunk.c
  chunk_adaptive.c
  chunk_constraint.c
  chunk_index.c
  chunk_scan.c
  constraint.c
  cross_module_fn.c
  copy.c
  compression_with_clause.c
  dimension.c
  dimension_slice.c
  dimension_vector.c
  estimate.c
  event_trigger.c
  extension.c
  extension_constants.c
  Generando código...
  Compilando...
  expression_utils.c
  foreign_key.c
  gapfill.c
  guc.c
  histogram.c
  hypercube.c
  hypertable.c
  hypertable_cache.c
  hypertable_restrict_info.c
  indexing.c
  init.c
  jsonb_utils.c
  license_guc.c
  osm_callbacks.c
  partitioning.c
  process_utility.c
  scanner.c
  scan_iterator.c
  sort_transform.c
  subspace_store.c
  Generando código...
  Compilando...
  timezones.c
  time_bucket.c
  time_utils.c
  custom_type_cache.c
  trigger.c
  utils.c
  version.c
  with_clause_parser.c
  tss_callbacks.c
  job.c
  job_stat.c
  job_stat_history.c
  launcher_interface.c
  scheduler.c
  timer.c
  conn.c
  conn_plain.c
  http.c
  http_response.c
  http_request.c
  Generando código...
  Compilando...
  conn_ssl.c
  functions.c
  replication.c
  stats.c
  telemetry_metadata.c
  telemetry.c
  policy.c
  chunk_stats.c
  array_utils.c
  catalog.c
  chunk_column_stats.c
  compression_chunk_size.c
  compression_settings.c
  continuous_agg.c
  continuous_aggs_watermark.c
  metadata.c
  tablespace.c
  allpaths.c
  ht_hypertable_modify.c
  list.c
  Generando código...
  Compilando...
  ts_explain.c
  hypertable_modify.c
  chunk_append.c
  exec.c
  transform.c
  chunk_dispatch.c
  chunk_insert_state.c
  constraint_aware_append.c
  add_hashagg.c
  constify_now.c
  constraint_cleanup.c
  expand_hypertable.c
  partialize.c
  space_constraint.c
  Generando código...
  agg_bookend.c
  planner.c
  planner.c
  planner.c
  agg_bookend.c
LINK : fatal error LNK1181: no se puede abrir el archivo de entrada 'no.obj' [D:\GIT\TIMESCALE\timescaledb\build\src\timescaledb.vcxproj]
  Building Custom Rule D:/GIT/TIMESCALE/timescaledb/src/loader/CMakeLists.txt
  loader.c
  bgw_message_queue.c
  bgw_counter.c
  bgw_launcher.c
  bgw_interface.c
  function_telemetry.c
  lwlocks.c
  Generando código...
LINK : fatal error LNK1181: no se puede abrir el archivo de entrada 'no.obj' [D:\GIT\TIMESCALE\timescaledb\build\src\loader\timescaledb-loader.vcxproj]
  Building Custom Rule D:/GIT/TIMESCALE/timescaledb/tsl/src/CMakeLists.txt
  chunk_api.c
  chunk.c
  chunkwise_agg.c
  init.c
  partialize_finalize.c
  process_utility.c
  reorder.c
  compression_api.c
  continuous_aggregate_api.c
  job.c
  job_api.c
  reorder_api.c
  retention_api.c
  policy_utils.c
  policies_v2.c
  api.c
  compression.c
  compression_dml.c
  compression_scankey.c
  compression_storage.c
  Generando código...
  Compilando...
  segment_meta.c
  array.c
  datum_serialize.c
  deltadelta.c
  dictionary.c
  gorilla.c
  common.c
  finalize.c
  insert.c
  invalidation_threshold.c
  invalidation.c
  materialize.c
  options.c
  refresh.c
  repair.c
  utils.c
  batch_array.c
  batch_queue_heap.c
  batch_queue_fifo.c
  compressed_batch.c
  Generando código...
  Compilando...
  decompress_chunk.c
  detoaster.c
  pred_text.c
  pred_vector_array.c
  qual_pushdown.c
  vector_predicates.c
  frozen_chunk_dml.c
  gapfill_functions.c
  gapfill_plan.c
  gapfill_exec.c
  locf.c
  interpolate.c
  functions.c
  grouping_policy_batch.c
  plan.c
  Generando código...
  planner.c
  create.c
  create.c
  planner.c
  exec.c
  planner.c
  planner.c
  exec.c
  exec.c
LINK : fatal error LNK1181: no se puede abrir el archivo de entrada 'no.obj' [D:\GIT\TIMESCALE\timescaledb\build\tsl\src\timescaledb-tsl.vcxproj]
```


### How can we reproduce the bug?

```bash
Follow the instrucctions to build from source on Windows from your website: https://docs.timescale.com/self-hosted/latest/install/installation-source/
```

There is a problem in the cmake project with the language of console messages.

These are the errors I was having:
`LINK : fatal error LNK1181: no se puede abrir el archivo de entrada 'no.obj' [D:\GIT\TIMESCALE\timescaledb\build\src\timescaledb.vcxproj]`

The problem was that vcxproj was being created incorrectly with the weird text "**no registrado**" :
`<AdditionalOptions>%(AdditionalOptions) /machine:x64 no registrado D:/Aplicaciones/PostgreSQL/16/lib/postgres.lib ws2_32.lib Version.lib</AdditionalOptions>`

This is because the terminal messages on my computer are in Spanish, and the pg_config command returns the text "no registrado" for some  variables, and  the CMakeLists is expecting the value "not recorded" (look at line 54).

So forcing English in the execution of pg_config in the cmake file solves the problem.

--- a/.unreleased/pr_7258
+++ b/.unreleased/pr_7258
@@ -1,0 +1,2 @@
+Fixes: #7258 Force English in the pg_config command executed by cmake to avoid unexpected building errors
+Thanks: @MiguelTubio for reporting and fixing a Windows build error

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,7 +41,7 @@ function(GET_PG_CONFIG var)
   # Only call pg_config if the variable didn't already have a value.
   if(NOT ${var})
     execute_process(
-      COMMAND ${PG_CONFIG} ${ARGN}
+      COMMAND ${CMAKE_COMMAND} -E env LC_MESSAGES=C ${PG_CONFIG} ${ARGN}
       OUTPUT_VARIABLE _temp
       OUTPUT_STRIP_TRAILING_WHITESPACE)
   endif()

